src/util.js

```diff
@@ -45,16 +67,24 @@ export function messageHandlerFunc(context, sqsQueue, handler) {
     const errorWrap = context.service?.wrapError || context.gb?.wrapError || (e => e);
 
     let parsedMessage;
+    let decodedMessage;
+    let decodedRest;
     try {
+      const contentEncoding = rest.MessageAttributes?.['Content-Encoding']?.StringValue;
       parsedMessage = JSON.parse(Body);
+      if (contentEncoding === supportedCompression) {
```

Comment:
If it's compressed { decompress, decode, parse  } , else { parse }

Only use parsedMessage